src/foundry/common/abstract/type-data.d.mts

```diff
@@ -81,21 +83,113 @@ declare class _InternalTypeDataModel<
 export default abstract class TypeDataModel<
   Schema extends DataSchema,
   Parent extends Document<DataSchema, any, any>,
-  BaseData extends Record<string, any> = Record<string, never>,
-  DerivedData extends Record<string, any> = Record<string, never>,
-> extends _InternalTypeDataModel<Schema, Parent, BaseData, DerivedData> {
+  DerivedData extends Record<string, unknown> = RemoveIndexSignatures<Record<string, never>>,
```

Comment:
I mentioned this but `RemoveIndexSignatures<Record<string, never>>` = `{}` which is unlikely to be what you want. Just leave it as `= Record<string, never>`; if the user _doesn't_ do the default you probably need remove index signatures anyways.